waspc/src/Wasp/Lib.hs

```diff
@@ -88,11 +90,17 @@ findWaspFile waspDir = do
       ".wasp" `isSuffixOf` SP.toFilePath path
         && (length (SP.toFilePath path) > length (".wasp" :: String))
 
-findDotEnvFile :: Path' Abs (Dir WaspProjectDir) -> IO (Maybe (Path' Abs File'))
-findDotEnvFile waspDir = do
-  let dotEnvAbsPath = waspDir SP.</> [relfile|.env|]
-  dotEnvExists <- doesFileExist (SP.toFilePath dotEnvAbsPath)
-  return $ if dotEnvExists then Just dotEnvAbsPath else Nothing
+findDotEnvServer :: Path' Abs (Dir WaspProjectDir) -> IO (Maybe (Path' Abs File'))
+findDotEnvServer waspDir = do
+  let dotEnvServerAbsPath = waspDir SP.</> [relfile|.env.server|]
+  dotEnvServerExists <- doesFileExist (SP.toFilePath dotEnvServerAbsPath)
+  return $ if dotEnvServerExists then Just dotEnvServerAbsPath else Nothing
+
+findDotEnvClient :: Path' Abs (Dir WaspProjectDir) -> IO (Maybe (Path' Abs File'))
+findDotEnvClient waspDir = do
+  let dotEnvClientAbsPath = waspDir SP.</> [relfile|.env.client|]
```

Comment:
I don't think we exported/used them like this before, but I wonder if now is a good time to have a single definition of `.env.server` and `.env.client`? We could put them in `ServerGenerator/Common.hs` and `WebAppGenerator/Common.hs` and then use those both here in `Lib.hs` as well as `ServerGenerator.hs` and `WebAppGenerator.hs`?

Reply:
Also, since `findDotEnvServer` and `findDotEnvClient` are identical (modulo the exact env file name), should we take that as a param and have a single function definition?

Reply:
> Also, since `findDotEnvServer` and `findDotEnvClient` are identical (modulo the exact env file name), should we take that as a param and have a single function definition?

I tried to create a general function that takes env type as a param... My Haskell skills are very limited. To do so, we have to put let's say `envType` variable inside  `[relfile|...]` expression. When I try `[relfile|envType]` it does not read variable and use its name instead.

It's pretty silly but that's what  my Haskell knowledge is

Reply:
Good job so far! One way you can DRY this up would be something like:

```haskell
findDotEnvServer :: Path' Abs (Dir WaspProjectDir) -> IO (Maybe (Path' Abs File'))
findDotEnvServer waspDir = toMaybeAbsFile $ waspDir SP.</> [relfile|.env.server|]

findDotEnvClient :: Path' Abs (Dir WaspProjectDir) -> IO (Maybe (Path' Abs File'))
findDotEnvClient waspDir = toMaybeAbsFile $ waspDir SP.</> [relfile|.env.client|]

toMaybeAbsFile :: Path' Abs File' -> IO (Maybe (Path' Abs File'))
toMaybeAbsFile dotEnvFile = do
  dotEnvFileExists <- doesFileExist (SP.toFilePath dotEnvFile)
  return $ if dotEnvFileExists then Just dotEnvFile else Nothing
```

The trick is playing with the types and seeing that this has the following type:
```haskell
waspDir SP.</> [relfile|.env.foo|] :: Path' Abs File'
```

This is ok, but we kinda lose some info here w.r.t. where files live. A slight improvement might be something like:

```haskell
findDotEnvServer :: Path' Abs (Dir WaspProjectDir) -> IO (Maybe (Path' Abs File'))
findDotEnvServer waspDir = toMaybeAbsFile waspDir [relfile|.env.server|]

findDotEnvClient :: Path' Abs (Dir WaspProjectDir) -> IO (Maybe (Path' Abs File'))
findDotEnvClient waspDir = toMaybeAbsFile waspDir [relfile|.env.client|]

toMaybeAbsFile :: Path' Abs (Dir WaspProjectDir) -> Path' (Rel WaspProjectDir) File' -> IO (Maybe (Path' Abs File'))
toMaybeAbsFile waspDir dotEnvFile = do
  let dotEnvAbsFp = waspDir SP.</> dotEnvFile
  dotEnvFileExists <- doesFileExist $ SP.toFilePath dotEnvAbsFp
  return $ if dotEnvFileExists then Just dotEnvAbsFp else Nothing
```

Here, we are saying our file is _relative_ to the wasp project dir. This also helps since we could then actually move those `relfile` bits into a shared declaration, again saying they are relative to that directory, and put it in some Common.hs file:

```haskell
dotEnvClient :: Path' (Rel WaspProjectDir) File'
dotEnvClient = [relfile|.env.client|]
```

Does that kinda make sense? StrongPath and Haskell typing does take a bit of time to grok, which is totally fine! :D

Reply:
Ah yes StrongPath is certainly a bit advanced thing, but it helps us a lot in the code - but don't worry about having a bit harder time grasping it!
Let us know if you want to do this change on your own -> if it feels too much, we will take care of it before we merge, and you can take a look at the result.

Reply:
Thanks @shayneczyzewski for providing those chunks of code. I injected the in the codebase

Reply:
So @Martinsos I've impended what Shayne proposes. If you still prefer it done differently let me know